metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -169,6 +169,19 @@ final class Diagnostics(
   def hasSyntaxError(path: AbsolutePath): Boolean =
     syntaxError.contains(path)
 
+  def hasDiagnosticError(path: AbsolutePath): Boolean = {
```

Comment:
Should we not check the build target errors? It will not work even if the file is ok, but some others failed.

Reply:
I went back and forth on this. If we want to be 100% sure they can actually organize the imports, then yea, we should check build target errors. However, there may be situations where they want to clean up a file and organize the imports in there and the semanticdb for that file is still valid since nothing changed in it. I could be convinced either way. What do you think?

Reply:
We could catch the stale exception and check afterwards if the build target contains an error. This would just mean changing the message in case of the stale semanticdb error.

Reply:
Ah yea that's a good idea. I'll add that extra check.

Reply:
I didn't know that we produce a semanticdb file if it's not compiling.